src/spicy/runtime-support.cc

```diff
@@ -221,6 +221,24 @@ TypePtr rt::event_arg_type(const EventHandlerPtr& handler, const hilti::rt::inte
     return zeek_args[idx];
 }
 
+zeek::analyzer::ID rt::current_analyzer_id() {
+    auto _ = hilti::rt::profiler::start("zeek/rt/current_analyzer_id");
+
+    if ( auto cookie = static_cast<Cookie*>(hilti::rt::context::cookie()) ) {
+        if ( auto x = cookie->protocol ) {
+            return x->analyzer->GetID();
+        }
+        else if ( auto x = cookie->file ) {
```

Comment:
Zeek's packet analysis analyzers don't set a current analyzer - e.g  [vxlan_packet](https://github.com/zeek/zeek/blob/991bc9644dd9b77f09fff992658f10500dce77e1/src/packet_analysis/protocol/vxlan/VXLAN.cc#L61-L67). For the one file analyzer we have (PE) there's also nothing obvious that sets the current analyzer.

Additionally, I'm slightly nervous that the IDs among the different analyzer kinds overlap. E.g. calling `disable_analyzer(c$id, current_analyzer())` within a packet analysis event (VXLAN, GENEVE, ...) or file analysis event would always attempt to disable a protocol analyzer which may by end-up working by accident.

So I'm starting to be conflicted if maybe returning 0 for non-protocol analyzers here makes more sense for the time being. Alternatively, whether protocol, packet and file analyzers should have separate ID spaces so that conflicts are impossible and one can tell by the numeric value what kind it is (similar to what we do with port spaces).

I'm approving anyhow as maybe I'm overthinking this.

Reply:
One problem is that the docs for `current_analyzer` don't specify what kind of analyzer it's returning an ID for:

```
## Returns the ID of the analyzer which raised the current event.
##
## Returns: The ID of the analyzer which raised the current event, or 0 if
##          none.
function current_analyzer%(%) : count
```

For protocol and file analyzers, there's only ever one instance of the analyzer so an ID is meaningful. The static counter could be moved up into the `zeek::id` namespace instead of having a separate counter for those two. 

For packet analyzers, the ID is set when the analyzer is instantiated. This means it's not really an ID for the type of analyzer but for that one instance. Is that really meaningful?